type-definitions/immutable.js.flow

```diff
@@ -464,7 +485,7 @@ declare class List<T> extends IndexedCollection<T> {
 }
 
 declare class Map<K,V> extends KeyedCollection<K,V> {
-  static <K, V>(): Map<K, V>;
+  static <K,V>(_: void): Map<K,V>;
```

Comment:
Type inference didn't work on objects without it. It looked like `flow` somehow ignores the object type, falling back to the constructor without named parameters. Since parameters in both constructors are optional, I thought that simply removing the empty constructor would solve the problem but it didn't. I added this one, and honestly I copied it from [here](https://github.com/facebook/flow/blob/master/lib/core.js#L424). It's hard for me to give further insight since I'm new to both `js` and `flow`.

I encourage to run the test without the patches first (to check out at d7ee48c) and then check out subsequent commits.